dms/models/ir_attachment.py

```diff
@@ -7,8 +8,10 @@ class IrAttachment(models.Model):
     _inherit = "ir.attachment"
 
     def _get_dms_directories(self, res_model, res_id):
-        return self.env["dms.directory"].search(
-            [("res_model", "=", res_model), ("res_id", "=", res_id)]
+        return (
+            self.env["dms.directory"]
+            .sudo()
```

Comment:
I don't think this is a good idea.

The method itself is not clear about it's doing this. I mean, you might be using in your module:

```py
self._get_dms_directories('res.partner', [1,2,3]).unlink()
```

... and expect to unlink only those directories that your current user has access to. The method is called `_get_dms_directories`, not `_get_dms_directories_sudo`, so the fact that the returned recordset is sudoed is, IMHO, an unexpected side effect and a gateway for security problems.

The correct fix would be, in this case, to prepend the `sudo()` _to the method call_:

```py
self.sudo()._get_dms_directories('res.partner', [1,2,3]).unlink()
```

In this case, all is predictable and expected.

Reply:
You approved the same in #53 ...

Reply:
I don't think it's too much problem, as it's an internal method and it wasn't renamed for not losing compatibility with inherited modules.

Reply:
Well, I didn't realize before. I still think this is a problem.

Reply:
Your proposal is not correct. In any case, it would be:

```python
self._get_dms_directories('res.partner', [1,2,3]).sudo().unlink()
```

but that won't get the same results, as we need that the search already contains all possible results.

Reply:
AFAIK, my proposal is correct 🤷🏼‍♂️ 

The sudo property is saved in the env, which is later used by the method, so it gets sudoed already.

Reply:
As a new handler is obtained with `self.env["dms.directory"]`, I would say the `sudo` is not kept, but anyway, please don't block this for this detail that doesn't impact and let's fix the underlying problem.

Reply:
I don't consider this a detail, because it can lead to security problems. However, to avoid blocking, I pushed the fix. Sometimes, it's faster to do it than to talk about it. 😃

Reply:
It's not a question of talking, but don't put an ugly code that we didn't want. My premise to Victor was: do the minimal `sudo` operations.

Reply:
I'm adding no more sudo operations. I'm just being explicit about them.

Reply:
3 against 1. You are adding more indeed.

Reply:
Those 3 were already being done inside the method, so there were 3 being done in any case. Readers just didn't realize.

Reply:
OK, do both what you think. I retire, but I don't want 2 commits for such simple thing.